shared/interfaces/metadata_interface.go

```diff
@@ -0,0 +1,17 @@
+package interfaces
+
+import (
+	"github.com/prysmaticlabs/go-bitfield"
+	pb "github.com/prysmaticlabs/prysm/proto/beacon/p2p/v1"
+)
+
+// Metadata returns the interface of a p2p metadata type.
+type Metadata interface {
```

Comment:
Do you want to add SyncBitfield() as well if you were to include V2 in the same PR

Reply:
For the general metadata interface it should not have anything from V2.

Reply:
I was thinking of adding this in our altair branch instead of here, given that is where it will be used.